### PR TITLE
feat: extend aws/cloudwatch with CloudWatch client

### LINKED ISSUE #958: Feature: extend aws/cloudwatch module with CloudWatch client
Having a CloudWatch client as part of the `aws/cloudwatch` module would be helpful for our tests.  

Any objections if I wrap up #229 by adding the missing automated tests?

### PR DESCRIPTION
**Pull request by patheard**
# Summary
This adds the ability to get a CloudWatch client that can be used to manage alarms, metrics and dashboards.

The code was originally created by @whage as part of #229. This adds the automated tests.

# Test results
```bash
go test -v -run TestNewCloudWatchClient
=== RUN   TestNewCloudWatchClient
=== PAUSE TestNewCloudWatchClient
=== RUN   TestNewCloudWatchClientE
=== PAUSE TestNewCloudWatchClientE
=== CONT  TestNewCloudWatchClient
=== CONT  TestNewCloudWatchClientE
TestNewCloudWatchClient 2021-07-27T15:08:59-04:00 region.go:92: Using region ap-south-1
TestNewCloudWatchClientE 2021-07-27T15:08:59-04:00 region.go:92: Using region eu-west-1
--- PASS: TestNewCloudWatchClient (0.00s)
--- PASS: TestNewCloudWatchClientE (0.00s)
PASS
ok      github.com/gruntwork-io/terratest/modules/aws   0.144s
```

# Related issues
Closes #229 
Closes #958

### CODE REVIEW

File: modules/aws/cloudwatch.go [hunk 1]
```diff
@@ -56,3 +57,21 @@ func NewCloudWatchLogsClientE(t testing.TestingT, region string) (*cloudwatchlog
 	}
 	return cloudwatchlogs.New(sess), nil
 }
+
+// NewCloudWatchClient creates a new CloudWatch client.
+func NewCloudWatchClient(t testing.TestingT, region string) *cloudwatch.CloudWatch {
+	client, err := NewCloudWatchClientE(t, region)
+	if err != nil {
+		t.Fatal(err)
+	}
```
Comment (brikis98) [inline modules/aws/cloudwatch.go:66]: NIT: we've started using `require.NoError(t, err)` to reduce these 3 lines down to 1.

**Review by brikis98 — APPROVED** [review]
LGTM! Just one tiny NIT and I can kick off tests.

File: modules/aws/cloudwatch.go (hunk 1 above)
Comment (patheard) [reply modules/aws/cloudwatch.go]: Sounds good - I've updated those blocks to use `require.NoError(t, err)`.

**Comment by patheard** [comment]
Updated test results with the `require.NoError(t, err)` change:
```
go test -v -run TestNewCloudWatchClient
=== RUN   TestNewCloudWatchClient
=== PAUSE TestNewCloudWatchClient
=== RUN   TestNewCloudWatchClientE
=== PAUSE TestNewCloudWatchClientE
=== CONT  TestNewCloudWatchClient
=== CONT  TestNewCloudWatchClientE
TestNewCloudWatchClient 2021-08-03T16:40:29-04:00 region.go:92: Using region ap-southeast-2
TestNewCloudWatchClientE 2021-08-03T16:40:29-04:00 region.go:92: Using region eu-west-3
--- PASS: TestNewCloudWatchClientE (0.00s)
--- PASS: TestNewCloudWatchClient (0.00s)
PASS
ok      github.com/gruntwork-io/terratest/modules/aws   0.365s
```

**Review by brikis98 — APPROVED** [review]
LGTM! I'll kick off tests now.

**Comment by rachelstowell** [comment]
Was this closed without being merged in?

**Comment by patheard** [comment]
Looks like and to be honest, I don't remember why 😓 

Feel free to copy my changes into a new PR the maintainers are still interested in them!

**Pull request closed without merging**